diff --git a/CMakeLists.txt b/CMakeLists.txt
index 213d5a8..eb80493 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,7 +166,7 @@ target_compile_options(${PROJECT_NAME}_objlib PRIVATE -fPIE -fPIC)
 if(CMAKE_COMPILER_IS_GNUCC AND CMAKE_C_COMPILER_VERSION VERSION_GREATER 6)
     target_compile_options(${PROJECT_NAME}_objlib PRIVATE -mmitigate-rop)
 endif()
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -z noexecstack -z relro -z now")
+# On MacOS, these flags aren't supported. Disable them to avoid build breakage
 
 option(BUILD_ERROR_ON_WARNING "Fail compilation on warning" OFF)
 
