diff --git a/include/safe_mem_lib.h b/include/safe_mem_lib.h
index 162efd2..a6059e8 100644
--- a/include/safe_mem_lib.h
+++ b/include/safe_mem_lib.h
@@ -97,7 +97,12 @@ extern errno_t wmemmove_s(wchar_t *dest, rsize_t dmax,
 
 
 /* set bytes */
+#ifdef __APPLE__
+// Use C standard definition
+#define memset_s(dest, dmax, value) memset_s(dest, dmax, value, dmax)
+#else
 extern errno_t memset_s(void *dest, rsize_t dmax, uint8_t value);
+#endif
 
 /* set uint16_t */
 extern errno_t memset16_s(uint16_t *dest, rsize_t dmax, uint16_t value);
